private/private.go

```diff
@@ -66,35 +68,119 @@ func MustNewPrivateTxManager(cfgPath string) PrivateTransactionManager {
-			Transport: unixTransport(cfg),
-		},
-		BaseURL: "http+unix://c",
+	var client *engine.Client
+	if engine.IsSocketConfigured(cfg) {
+		log.Info("Connecting to private tx manager using IPC socket")
+		client = createIPCClient(cfg)
+	} else if engine.IsTlsConfigured(cfg) {
+		log.Info("Connecting to private tx manager using HTTPS")
+		client, err = createHTTPClientUsingTLS(cfg)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create http.client to private tx manager using '%s' due to: %s", cfgPath, err)
+		}
+	} else {
+		log.Info("Connecting to private tx manager using HTTP")
+		client = createHTTPClient(cfg)
 	}
 
 	ptm, err := selectPrivateTxManager(client)
 	if err != nil {
-		return nil, fmt.Errorf("unable to connect to private tx manager using %s due to %s", cfgPath, err)
+		return nil, fmt.Errorf("unable to connect to private tx manager using '%s' due to: %s", cfgPath, err)
 	}
 	return ptm, nil
 }
 
+func createIPCClient(cfg engine.Config) *engine.Client {
+	client := &engine.Client{
+		HttpClient: &http.Client{
+			Transport: unixTransport(cfg),
+		},
+		BaseURL: "http+unix://c",
+	}
+	return client
+}
+
 func unixTransport(cfg engine.Config) *httpunix.Transport {
 	t := &httpunix.Transport{
-		DialTimeout:           time.Duration(cfg.DialTimeout) * time.Second,
-		RequestTimeout:        time.Duration(cfg.RequestTimeout) * time.Second,
-		ResponseHeaderTimeout: time.Duration(cfg.ResponseHeaderTimeout) * time.Second,
+		DialTimeout:           time.Duration(cfg.SocketConfig.DialTimeout) * time.Second,
+		RequestTimeout:        time.Duration(cfg.SocketConfig.RequestTimeout) * time.Second,
+		ResponseHeaderTimeout: time.Duration(cfg.SocketConfig.ResponseHeaderTimeout) * time.Second,
 	}
-	t.RegisterLocation("c", filepath.Join(cfg.WorkDir, cfg.Socket))
+	t.RegisterLocation("c", filepath.Join(cfg.SocketConfig.WorkDir, cfg.SocketConfig.Socket))
 	return t
 }
 
+func createHTTPClient(cfg engine.Config) *engine.Client {
+	client := &engine.Client{
+		HttpClient: &http.Client{
+			Timeout:   time.Duration(cfg.HttpConfig.ClientTimeout) * time.Second,
+			Transport: httpTransport(cfg),
+		},
+		BaseURL: cfg.HttpConfig.Url,
+	}
+	return client
+}
+
+func httpTransport(cfg engine.Config) *http.Transport {
+	t := &http.Transport{
+		IdleConnTimeout: time.Duration(cfg.HttpConfig.IdleConnTimeout) * time.Second,
+		WriteBufferSize: cfg.HttpConfig.WriteBufferSize,
+		ReadBufferSize:  cfg.HttpConfig.ReadBufferSize,
+	}
+	return t
+}
+
+func createHTTPClientUsingTLS(cfg engine.Config) (*engine.Client, error) {
+	transport, err := httpTransportUsingTLS(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &engine.Client{
+		HttpClient: &http.Client{
+			Timeout:   time.Duration(cfg.HttpConfig.ClientTimeout) * time.Second,
+			Transport: transport,
+		},
+		BaseURL: cfg.HttpConfig.Url,
+	}
+	return client, nil
+}
+
+func httpTransportUsingTLS(cfg engine.Config) (*http.Transport, error) {
+	rootCAPool := x509.NewCertPool()
```

Comment:
Would it not be possible to implement a single `CreateHttpTransport` function doing the job for both TLS and non-TLS? (would avoid some code duplication)

Reply:
Done